nj2k/src/org/jetbrains/kotlin/nj2k/JavaToJKTreeBuilder.kt

```diff
@@ -735,18 +736,23 @@ class JavaToJKTreeBuilder constructor(
             }
         }
 
-        fun <T : PsiModifierListOwner> T.annotationList(docCommentOwner: PsiDocCommentOwner?): JKAnnotationList {
+        fun <T : PsiModifierListOwner> T.annotationList(
+            docCommentOwner: PsiDocCommentOwner?,
+            typeAnnotations: Array<out PsiAnnotation>
+        ): JKAnnotationList {
             val deprecatedAnnotation = docCommentOwner?.docComment?.deprecatedAnnotation()
             val plainAnnotations = annotations.mapNotNull { annotation ->
                 when {
                     annotation !is PsiAnnotation -> null
                     annotation.qualifiedName == DEPRECATED_ANNOTAION_FQ_NAME && deprecatedAnnotation != null -> null
+                    annotation in typeAnnotations -> null
```

Comment:
`it.owner` returns PsiModifierList, not PsiTypeElement. Could you tell me how to filter out by `it.owner`?

Reply:
As far, as I can see `com.intellij.psi.PsiAnnotation#getOwner` has type `PsiAnnotationOwner` and `PsiTypeElement` is subtype of it

Reply:
I think `getOwner()` returns `PsiModifierList` here:
https://github.com/JetBrains/intellij-community/blob/master/java/java-psi-impl/src/com/intellij/psi/impl/source/tree/java/PsiAnnotationImpl.java#L120-L124

Reply:
You are right :) then `com.intellij.codeInsight.AnnotationTargetUtil#isTypeAnnotation` is your friend here